aiida_zeopp/data/parameters.py

```diff
@@ -140,15 +143,23 @@ def output_dict(self):
         Keys are the selected options that require an output file name,
         values are the file names.
         """
-        return {
-            k: self._OUTPUT_FILE_PREFIX.format(k)
-            for k in self.get_dict() if k in list(output_options.keys())
-        }
+        d = {}
+        for k in self.get_dict():
+            if k in list(output_options.keys()):
+                if k != 'visVoro':
+                    d.update({k: self._OUTPUT_FILE_PREFIX.format(k)})
+                elif k == 'visVoro':
+                    d.update({
+                        'visVoro_allnodes' : self._OUTPUT_FILE_PREFIX.format(k) + '_voro.xyz',
+                        'visVoro_accnodes' : self._OUTPUT_FILE_PREFIX.format(k) + '_voro_accessible.xyz',
+                        'visVoro_nonaccnodes' : self._OUTPUT_FILE_PREFIX.format(k) + '_voro_nonaccessible.xyz',
+                    })
```

Comment:
same here as above (exception disappears)

Reply:
In this case, I am a bit puzzled. Part of the output names is different. Two solutions came to my mind:
1) It still would contain introducing an exception but on the length of label list rather the key string.
2) Modifying the `zeo++` itself to generate the output files with an index rather different strings.

Reply:
You're right.

So, if your parameter labels were `visvoro_voro`, `visvoro_accessible` and `visvoro_nonaccessible` (which I anyhow recommend) and if zeo++ would use <prefix>.visvoro_voro, <prefix>.visvoro_accessible, and <prefix>.visvoro_nonaccessible to write out the files, it would work.

However, there would be no clean way to maintain backward compatibility with the old zeo++ file naming...
Either that has to go or we need to make an exception here.

I leave this decision up to you.

Reply:
Edit: Changed the names in my "recommendation" ;-)

Reply:
Thanks for suggestions. 
I have modified the `zeo++` to generate the output files with the above-mentioned naming. It happens only if we provide a `<prefix>`. Then, I think we are good with backward compatibility of `zeo++`.

In the case of plugin, so far I came up with the following solution for the `output_dict` function:
```
    @property
    def output_dict(self):
        """Return dictionary of specified options requiring an output file name.

        Keys are the selected options that require an output file name,
        values are the file names.
        """
        output_dict = {}
        output_files = []
        for k,v in output_options.items():
            if len(v[1]) > 1 :
                output_dict[k] = {}
                for i in range(len(v[1])):
                    output_dict.update({k + str(i+1) : self._OUTPUT_FILE_PREFIX.format(k + str(v[1][i]))})
                    output_files.append(self._OUTPUT_FILE_PREFIX.format(k + str(v[1][i])))
            else:
                output_dict.update({k : self._OUTPUT_FILE_PREFIX.format(k)})
                output_files.append(self._OUTPUT_FILE_PREFIX.format(k))
        return output_dict, output_files


```
which for instance generates the the `output_dict` as:
```
{'cssr': 'out.cssr',
 'v1': 'out.v1',
 'xyz': 'out.xyz',
 'nt2': 'out.nt2',
 'res': 'out.res',
 'zvis': 'out.zvis',
 'axs': 'out.axs',
 'visVoro': {'visVoro1': 'out.voro',
  'visVoro2': 'out.voro_accessible',
  'visVoro3': 'out.voro_nonaccessible'}}
```
and the `output_file` list as:
```
['out.cssr',
 'out.v1',
 'out.xyz',
 'out.nt2',
 'out.res',
 'out.zvis',
 'out.axs',
 'out.voro',
 'out.voro_accessible',
 'out.voro_nonaccessible']
```
Then, we even can remove the  `output_keys` and `output_files` functions and access the required list and keys from `output_dict(output_options)[0].keys()` and `output_dict(output_options)[1]`.

Reply:
Hi @ltalirz 
I could get back to this PR and finalize the changes. I already pushed them so we can discuss on improving and finalizing the changes. 
I have tried different scenarios to prevent introducing exceptions based on your useful comments above. 
Currently, 
* there is only once exception in `def output_dict(self):` as I could not come up with a solution to rescue the filenames for `visVoro`
* I needed to add a property function to fix the problem with defining appropriate prefix in command line arguments. It caused a little change in the code. 

Please let me know your opinion about the changes and if there are other things that need to be taken care of.